tests/Integration/Behaviour/Features/Scenario/CartRule/cart_rule.feature

```diff
@@ -0,0 +1,262 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s cart_rule
+@reset-database-before-feature
+Feature: Add cart rule
+  PrestaShop allows BO users to create cart rules
+  As a BO user
+  I must be able to create cart rules
+
+  Background:
+    Given shop "shop1" with name "test_shop" exists
+    Given there is a currency named "currency1" with iso code "USD" and exchange rate of 0.92
+    Given there is a currency named "currency2" with iso code "CHF" and exchange rate of 1.25
+    Given currency "currency1" is the default one
+
+  Scenario: Create a cart rule with amount discount
+    When I want to create a new cart rule
```

Comment:
Because this scenario wasn't written by me, it was done before in tests/Integration/Behaviour/Features/Scenario/CartRule/add_cart_rule.feature file to test cart rule creation. I simply moved it to cart_rule.feature. 

I needed to create cart rule as well however this method wasn't fitting for me because it doesn't store cart rules reference and I needed reference to delete cart rule later(and check it). So I created new function which creates minimal cart rule that stores reference.

In a hindsight maybe I should have altered this scenario to store cart rules reference instead.

Reply:
So your version is the best 😄 (or at least the one I prefer)
Let's no make the scope of this PR bigger so let's keep it like this for now.

Could yo do only 1 thing ? please put a `@todo` comment on the `Scenario: Create a cart rule with amount discount` (and the other scenarios which describe every property one by one) so that in the future, when eligible*, we use your versions (with the TableNode) instead. It's much easier to read.

Having a Behat step "I set property X of object Y to Z" is fine for small, user-oriented scenario (tables make them harder to read) but here it just becomes a giant list 😄

Reply:
Just so I don't have to rewrite the todo multiple times, I wanna confirm it here first.
Do you mean like this?

@todo If you need to create cart rule, use "I create cart rule "cart_rule_1" with the following properties" instead from Delete cart rule scenario instead.